### PR TITLE
[10-10EZ] Update specs to improve test coverage

### PR DESCRIPTION
**Pull request by coope93**
## Summary

- This work is behind a feature toggle (flipper): NO
- Added additional specs for a logic branch in the `HealthCareApplication` model and in the `StdInstitutionImportJob`.
- Deleted an unused method (`form_id`) that was not being tested in `HealthCareApplication`. We already have that value in a const in the model and it is already uppercase. This was the only implementation change. 
- 10-10 Health Apps

## Testing done

- [x] *New code is covered by unit tests*

## What areas of the site does it impact?
10-10EZ Form

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [x]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

I searched for any usages of form_id and am expecting our specs to fail if there is somewhere we are using it. But if anyone knows of somewhere we use it obviously we don't want to remove it.

### CODE REVIEW

**Review by Mitch-A6 — DISMISSED** [review]
nice cleanup! 🔥

File: spec/models/health_care_application_spec.rb [hunk 1]
```diff
@@ -189,9 +188,15 @@
...
+      it 'returns partial ee data when enrollment_status is present' do
         expect(described_class.parsed_ee_data(ee_data, false)).to eq(
-          parsed_status: login_required
+          parsed_status: HCA::EnrollmentEligibility::Constants::LOGIN_REQUIRED
+        )
+      end
+
+      it 'returns none of the above ee data when enrollment_status is not set' do
+        expect(described_class.parsed_ee_data({}, false)).to eq(
+          parsed_status: HCA::EnrollmentEligibility::Constants::NONE_OF_THE_ABOVE
```
Comment (stevenjcumming) [inline spec/models/health_care_application_spec.rb:199]: Could refactor these to examples into this, just to increase consistency with the spec a bit

when enrollment_status is present

```ruby
context 'when enrollment_status is present' do 
  it 'returns partial ee data' .... 
end

context 'when enrollment_status is not set' do 
  it 'returns none of the above ee data' ..
end
```

Comment (coope93) [reply spec/models/health_care_application_spec.rb]: Yeah, I agree that makes it read more clearly. I'll update it 👍

**Review by Mitch-A6 — APPROVED** [review]
👍 love the added coverage

**Review by stevenjcumming — APPROVED** [review]

**Pull request merged**